src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -585,11 +577,26 @@ internal val deleteEntitySetEntityKeys = "DELETE FROM ${IDS.name} WHERE ${ENTITY
  *
  * 1. entity set id
  * 2. entity key ids
+ *    IF LINKING    checks against ORIGIN_ID
+ *    ELSE          checks against ID column
  * 3. partition
  * 4. property type ids
  */
-internal val deletePropertiesOfEntitiesInEntitySet = "DELETE FROM ${DATA.name} " +
-        "WHERE ${ENTITY_SET_ID.name} = ? AND ${ID_VALUE.name} = ANY(?) AND ${PARTITION.name} = ? AND ${PROPERTY_TYPE_ID.name} = ANY(?) "
+internal fun deletePropertiesOfEntitiesInEntitySet(linking: Boolean = false): String {
+    val idsColumn = if (linking) {
```

Comment:
Is it not the case that when a non-linking entity has properties deleted that the linking entity should also have those properties deleted?

Reply:
Yes, you are right, but this is done in two different transactions, one called with `linking = false` and one with `linking = true`, so we check the non-linking entity key ids against the `ID` column and then against `ORIGIN_ID` column.  If you look at the code here, you will see that we execute this query twice: https://github.com/openlattice/conductor-client/pull/761/files#diff-9eff35dfabde14ebc448dc06ae0893f2R681

Reply:
Awesome :)